rxbindings/src/main/java/com/amplifyframework/rx/RxApiBinding.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.rx;
+
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.api.ApiCategoryBehavior;
+import com.amplifyframework.api.ApiException;
+import com.amplifyframework.api.graphql.GraphQLRequest;
+import com.amplifyframework.api.graphql.GraphQLResponse;
+import com.amplifyframework.api.graphql.MutationType;
+import com.amplifyframework.api.graphql.SubscriptionType;
+import com.amplifyframework.api.rest.RestOptions;
+import com.amplifyframework.api.rest.RestResponse;
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.core.model.Model;
+import com.amplifyframework.core.model.query.predicate.QueryPredicate;
+
+import io.reactivex.Observable;
+import io.reactivex.Single;
+
+/**
+ * An implementation of the RxApi which satisfies the API contract by wrapping
+ * {@link ApiCategoryBehavior} in Rx primitives.
+ */
+final class RxApiBinding implements RxApi {
+    private final ApiCategoryBehavior api;
+
+    RxApiBinding() {
+        this(Amplify.API);
+    }
+
+    @SuppressWarnings("WeakerAccess")
+    RxApiBinding(ApiCategoryBehavior api) {
```

Comment:
Is this meant to be for plugin or category?

Reply:
For the category’s APIs. This stuff should work regardless of which plugin is implementing the behaviors.

Reply:
Then I think `ApiCategory` is a better fit than `ApiCategoryBehavior`, since `RxApiBinding` simulates the category, not just a plugin

Reply:
Hmm, yes you might be right. I think I agree. Let me give it a try anyway and see if there are any negative side effects to that proposal.

Reply:
It worked, I just had to play some small games in the unit tests to deal with the `final` categories.